sycl/doc/design/CompilerAndRuntimeDesign.md

```diff
@@ -682,7 +683,7 @@ on the following logic:
 
 - If the 2 versions exist, the original kernel is called if global
   offset is 0 otherwise it will call the cloned one and pass the
-  offset by value;
+  offset by value (for CUDA backend), or by ref for AMD;
```

Comment:
Is this really a difference between the CUDA and AMD backends?  When I look at the example LLVM IR code listing below (which predated your change), it shows the kernel taking a pointer to the offset even for CUDA and that pointer argument has the `byval` attribute.  If AMD is different, how is it different?

Reply:
Is there a difference in how the runtime passes the offset for the two backends?  I was assuming that it just calls `clSetKernelArg` / `zeKernelSetArgumentValue` regardless of the backend.  Is that not the case?

Reply:
I doubt this will ever go down the OpenCL path, but even if it did, global offsets are passed as implicit arguments to the kernel, users will not be able to set it directly. This is more to do with how/what the compiler backend can handle.